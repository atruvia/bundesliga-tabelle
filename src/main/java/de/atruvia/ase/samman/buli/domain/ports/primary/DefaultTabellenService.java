package de.atruvia.ase.samman.buli.domain.ports.primary;

import java.util.List;

import org.springframework.stereotype.Service;

import de.atruvia.ase.samman.buli.domain.Paarung;
import de.atruvia.ase.samman.buli.domain.Tabelle;
import de.atruvia.ase.samman.buli.domain.TabellenPlatz;
import de.atruvia.ase.samman.buli.domain.ports.secondary.SpieltagRepo;
import de.atruvia.ase.samman.buli.domain.ports.secondary.WappenRepo;
import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
class DefaultTabellenService implements TabellenService {

	private final SpieltagRepo spieltagRepo;
	private final WappenRepo wappenRepo;

	@Override
	public List<TabellenPlatz> erstelleTabelle(String league, String season) {
		Tabelle tabelle = new Tabelle(wappenRepo);
		lade(league, season).forEach(tabelle::add);
		return tabelle.getEntries();
	}

	private List<Paarung> lade(String league, String season) {
		try {
			return spieltagRepo.lade(league, season);
		} catch (Exception e) {
			throw new RuntimeException(e);
		}
	}

}
