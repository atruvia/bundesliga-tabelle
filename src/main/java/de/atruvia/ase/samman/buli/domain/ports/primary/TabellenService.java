package de.atruvia.ase.samman.buli.domain.ports.primary;

import java.util.List;

import de.atruvia.ase.samman.buli.domain.TabellenPlatz;

public interface TabellenService {

	public interface Tabelle {
		String name();

		List<TabellenPlatz> eintraege();
	}

	Tabelle erstelleTabelle(String league, String season);

}
