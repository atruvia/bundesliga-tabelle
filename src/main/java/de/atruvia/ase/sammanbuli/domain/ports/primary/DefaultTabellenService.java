package de.atruvia.ase.sammanbuli.domain.ports.primary;

import de.atruvia.ase.sammanbuli.domain.ports.secondary.SpieltagRepo;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class DefaultTabellenService implements TabellenService {

	private final SpieltagRepo spieltagRepo;

	@Override
	public void erstelleTabelle(String league, String season) {
		Tabelle
		
		
		// TODO Auto-generated method stub

	}

}
