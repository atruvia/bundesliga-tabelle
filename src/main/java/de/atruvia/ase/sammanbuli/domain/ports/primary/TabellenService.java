package de.atruvia.ase.sammanbuli.domain.ports.primary;

import java.util.List;

import de.atruvia.ase.samman.buli.domain.TabellenPlatz;

public interface TabellenService {

	List<TabellenPlatz> erstelleTabelle(String league, String season);

}
