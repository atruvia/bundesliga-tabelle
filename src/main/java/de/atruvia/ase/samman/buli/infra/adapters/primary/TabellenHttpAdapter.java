package de.atruvia.ase.samman.buli.infra.adapters.primary;

import java.util.List;
import java.util.stream.Stream;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

import de.atruvia.ase.samman.buli.domain.Paarung.Ergebnis;
import de.atruvia.ase.samman.buli.domain.TabellenPlatz;
import de.atruvia.ase.samman.buli.domain.ports.primary.TabellenService;
import lombok.Builder;
import lombok.RequiredArgsConstructor;
import lombok.Value;

@RestController
@RequiredArgsConstructor
public class TabellenHttpAdapter {

	@Value
	@Builder
	private static class JsonTabellenPlatz {

		int platz;
		String wappen;
		String team;
		int spiele;
		int punkte;
		int tore, gegentore, tordifferenz;
		int siege, unentschieden, niederlagen;
		String letzte5;

		private static JsonTabellenPlatz fromDomain(TabellenPlatz domain) {
			return builder() //
					.platz(domain.getPlatz()) //
					.wappen(domain.getWappen() == null ? null : domain.getWappen().toASCIIString()) //
					.team(domain.getTeam()) //
					.spiele(domain.getSpiele()) //
					.punkte(domain.getPunkte()) //
					.tore(domain.getTore()) //
					.gegentore(domain.getGegentore()) //
					.tordifferenz(domain.getTorDifferenz()) //
					.siege(domain.getSiege()) //
					.unentschieden(domain.getUnentschieden()) //
					.niederlagen(domain.getNiederlagen()) //
					.letzte5(convert(domain)) //
					.build();
		}

		private static List<Ergebnis> convert(TabellenPlatz domain) {
			return domain.getLetzte(5);
		}

	}

	private final TabellenService tabellenService;

	@GetMapping("/tabelle/{league}/{season}")
	public Stream<JsonTabellenPlatz> getTabelle(@PathVariable String league, @PathVariable String season) {
		return tabellenService.erstelleTabelle(league, season).stream().map(JsonTabellenPlatz::fromDomain);
	}

}
