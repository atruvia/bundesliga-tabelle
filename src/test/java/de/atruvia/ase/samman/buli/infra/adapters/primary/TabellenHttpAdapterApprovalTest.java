package de.atruvia.ase.samman.buli.infra.adapters.primary;

import static de.atruvia.ase.samman.buli.infra.adapters.secondary.OldbSpieltagRepoMother.spieltagFsRepo;
import static org.approvaltests.JsonApprovals.verifyJson;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Import;
import org.springframework.test.web.servlet.MockMvc;

import de.atruvia.ase.samman.buli.domain.ports.primary.DefaultTabellenService;
import de.atruvia.ase.samman.buli.domain.ports.primary.TabellenService;

@WebMvcTest
@AutoConfigureMockMvc
@Import(TabellenHttpAdapterApprovalTest.OldbConfig.class)
class TabellenHttpAdapterApprovalTest {

	@Autowired
	MockMvc mockMvc;

	@Test
	void approveWithRunningGames() throws Exception {
		String league = "bl1";
		String season = "2023-games-running";
		String jsonResponse = mockMvc.perform(get("/tabelle/" + league + "/" + season)) //
				.andDo(print()) //
				.andReturn() //
				.getResponse() //
				.getContentAsString();
		verifyJson(jsonResponse);
	}

	@TestConfiguration
	static class OldbConfig {
		@Bean
		TabellenService tabellenService() {
			return new DefaultTabellenService(spieltagFsRepo());
		}
	}
}
