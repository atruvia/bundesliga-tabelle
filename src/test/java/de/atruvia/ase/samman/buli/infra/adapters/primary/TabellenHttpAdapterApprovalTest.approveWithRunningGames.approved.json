[
  {
    "platz": 1,
    "wappen": "https://upload.wikimedia.org/wikipedia/de/b/b3/Fc_st_pauli_logo.svg",
    "team": "FC St. Pauli",
    "spiele": 26,
    "punkte": 54,
    "tore": 48,
    "gegentore": 25,
    "tordifferenz": 23,
    "siege": 15,
    "unentschieden": 9,
    "niederlagen": 2,
    "letzte5": "SSNSS",
    "tendenz": [
      "S",
      "S",
      "N",
      "S",
      "S"
    ]
  },
  {
    "platz": 2,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/thumb/3/30/Holstein_Kiel_Logo.svg/300px-Holstein_Kiel_Logo.svg.png",
    "team": "Holstein Kiel",
    "spiele": 27,
    "punkte": 52,
    "tore": 50,
    "gegentore": 34,
    "tordifferenz": 16,
    "siege": 16,
    "unentschieden": 4,
    "niederlagen": 7,
    "letzte5": "SSUNS",
    "tendenz": [
      "S",
      "S",
      "U",
      "N",
      "S"
    ],
    "laufendesSpiel": {
      "ergebnis": "S",
      "gegner": "FC Hansa Rostock",
      "tore": 1,
      "toreGegner": 0
    }
  },
  {
    "platz": 3,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/f/f7/Hamburger_SV_logo.svg",
    "team": "Hamburger SV",
    "spiele": 26,
    "punkte": 44,
    "tore": 50,
    "gegentore": 37,
    "tordifferenz": 13,
    "siege": 13,
    "unentschieden": 5,
    "niederlagen": 8,
    "letzte5": "SNNSU",
    "tendenz": [
      "S",
      "N",
      "N",
      "S",
      "U"
    ]
  },
  {
    "platz": 4,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/thumb/9/94/Fortuna_D%C3%BCsseldorf.svg/150px-Fortuna_D%C3%BCsseldorf.svg.png",
    "team": "Fortuna DÃ¼sseldorf",
    "spiele": 27,
    "punkte": 43,
    "tore": 56,
    "gegentore": 35,
    "tordifferenz": 21,
    "siege": 12,
    "unentschieden": 7,
    "niederlagen": 8,
    "letzte5": "SSUSU",
    "tendenz": [
      "S",
      "S",
      "U",
      "S",
      "U"
    ],
    "laufendesSpiel": {
      "ergebnis": "N",
      "gegner": "1. FC Kaiserslautern",
      "tore": 0,
      "toreGegner": 1
    }
  },
  {
    "platz": 5,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/c/cd/Hannover_96_Logo.svg",
    "team": "Hannover 96",
    "spiele": 26,
    "punkte": 40,
    "tore": 47,
    "gegentore": 35,
    "tordifferenz": 12,
    "siege": 10,
    "unentschieden": 10,
    "niederlagen": 6,
    "letzte5": "UUUNS",
    "tendenz": [
      "U",
      "U",
      "U",
      "N",
      "S"
    ]
  },
  {
    "platz": 6,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/e/e3/SC_Paderborn_07_Logo.svg",
    "team": "SC Paderborn 07",
    "spiele": 26,
    "punkte": 39,
    "tore": 40,
    "gegentore": 43,
    "tordifferenz": -3,
    "siege": 11,
    "unentschieden": 6,
    "niederlagen": 9,
    "letzte5": "NUUSN",
    "tendenz": [
      "N",
      "U",
      "U",
      "S",
      "N"
    ]
  },
  {
    "platz": 7,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/c/c8/Karlsruher_SC_Logo_2.svg",
    "team": "Karlsruher SC",
    "spiele": 26,
    "punkte": 38,
    "tore": 55,
    "gegentore": 41,
    "tordifferenz": 14,
    "siege": 10,
    "unentschieden": 8,
    "niederlagen": 8,
    "letzte5": "SNSSU",
    "tendenz": [
      "S",
      "N",
      "S",
      "S",
      "U"
    ]
  },
  {
    "platz": 8,
    "wappen": "https://i.imgur.com/pwuVbKr.png",
    "team": "SpVgg Greuther FÃ¼rth",
    "spiele": 26,
    "punkte": 38,
    "tore": 37,
    "gegentore": 38,
    "tordifferenz": -1,
    "siege": 11,
    "unentschieden": 5,
    "niederlagen": 10,
    "letzte5": "NNNSN",
    "tendenz": [
      "N",
      "N",
      "N",
      "S",
      "N"
    ]
  },
  {
    "platz": 9,
    "wappen": "https://i.imgur.com/apFwbYZ.png",
    "team": "Hertha BSC",
    "spiele": 26,
    "punkte": 37,
    "tore": 50,
    "gegentore": 43,
    "tordifferenz": 7,
    "siege": 10,
    "unentschieden": 7,
    "niederlagen": 9,
    "letzte5": "SNUUS",
    "tendenz": [
      "S",
      "N",
      "U",
      "U",
      "S"
    ]
  },
  {
    "platz": 10,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/d/d4/SV_Elversberg_Logo_2015.svg",
    "team": "SV Elversberg 07",
    "spiele": 27,
    "punkte": 36,
    "tore": 39,
    "gegentore": 43,
    "tordifferenz": -4,
    "siege": 10,
    "unentschieden": 6,
    "niederlagen": 11,
    "letzte5": "NSNNS",
    "tendenz": [
      "N",
      "S",
      "N",
      "N",
      "S"
    ],
    "laufendesSpiel": {
      "ergebnis": "U",
      "gegner": "Eintracht Braunschweig",
      "tore": 0,
      "toreGegner": 0
    }
  },
  {
    "platz": 11,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/f/fa/1._FC_N%C3%BCrnberg_logo.svg",
    "team": "1. FC NÃ¼rnberg",
    "spiele": 26,
    "punkte": 36,
    "tore": 35,
    "gegentore": 45,
    "tordifferenz": -10,
    "siege": 10,
    "unentschieden": 6,
    "niederlagen": 10,
    "letzte5": "NSSNU",
    "tendenz": [
      "N",
      "S",
      "S",
      "N",
      "U"
    ]
  },
  {
    "platz": 12,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/d/d3/Logo_1_FC_Kaiserslautern.svg",
    "team": "1. FC Kaiserslautern",
    "spiele": 27,
    "punkte": 32,
    "tore": 43,
    "gegentore": 51,
    "tordifferenz": -8,
    "siege": 9,
    "unentschieden": 5,
    "niederlagen": 13,
    "letzte5": "USSNU",
    "tendenz": [
      "U",
      "S",
      "S",
      "N",
      "U"
    ],
    "laufendesSpiel": {
      "ergebnis": "S",
      "gegner": "Fortuna DÃ¼sseldorf",
      "tore": 1,
      "toreGegner": 0
    }
  },
  {
    "platz": 13,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/thumb/8/84/1._FC_Magdeburg.svg/210px-1._FC_Magdeburg.svg.png",
    "team": "1. FC Magdeburg",
    "spiele": 26,
    "punkte": 31,
    "tore": 38,
    "gegentore": 41,
    "tordifferenz": -3,
    "siege": 8,
    "unentschieden": 7,
    "niederlagen": 11,
    "letzte5": "NNUSN",
    "tendenz": [
      "N",
      "N",
      "U",
      "S",
      "N"
    ]
  },
  {
    "platz": 14,
    "wappen": "https://upload.wikimedia.org/wikipedia/de/thumb/3/3d/Logo_SV_Wehen_Wiesbaden.svg/1024px-Logo_SV_Wehen_Wiesbaden.svg.png",
    "team": "SV Wehen Wiesbaden",
    "spiele": 26,
    "punkte": 31,
    "tore": 30,
    "gegentore": 34,
    "tordifferenz": -4,
    "siege": 8,
    "unentschieden": 7,
    "niederlagen": 11,
    "letzte5": "NUSNN",
    "tendenz": [
      "N",
      "U",
      "S",
      "N",
      "N"
    ]
  },
  {
    "platz": 15,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/9/97/FC_Schalke_04_Logo.png",
    "team": "FC Schalke 04",
    "spiele": 26,
    "punkte": 30,
    "tore": 42,
    "gegentore": 54,
    "tordifferenz": -12,
    "siege": 9,
    "unentschieden": 3,
    "niederlagen": 14,
    "letzte5": "NUSNS",
    "tendenz": [
      "N",
      "U",
      "S",
      "N",
      "S"
    ]
  },
  {
    "platz": 16,
    "wappen": "https://upload.wikimedia.org/wikipedia/de/4/45/Logo_Eintracht_Braunschweig.svg",
    "team": "Eintracht Braunschweig",
    "spiele": 27,
    "punkte": 28,
    "tore": 25,
    "gegentore": 39,
    "tordifferenz": -14,
    "siege": 8,
    "unentschieden": 4,
    "niederlagen": 15,
    "letzte5": "SNNUN",
    "tendenz": [
      "S",
      "N",
      "N",
      "U",
      "N"
    ],
    "laufendesSpiel": {
      "ergebnis": "U",
      "gegner": "SV Elversberg 07",
      "tore": 0,
      "toreGegner": 0
    }
  },
  {
    "platz": 17,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/0/00/FC_Hansa_Logo_seit_2009.png",
    "team": "FC Hansa Rostock",
    "spiele": 27,
    "punkte": 28,
    "tore": 24,
    "gegentore": 42,
    "tordifferenz": -18,
    "siege": 8,
    "unentschieden": 4,
    "niederlagen": 15,
    "letzte5": "SSNNU",
    "tendenz": [
      "S",
      "S",
      "N",
      "N",
      "U"
    ],
    "laufendesSpiel": {
      "ergebnis": "N",
      "gegner": "Holstein Kiel",
      "tore": 0,
      "toreGegner": 1
    }
  },
  {
    "platz": 18,
    "wappen": "https://upload.wikimedia.org/wikipedia/commons/2/24/Logo_Vfl_Osnabrueck_2017.png",
    "team": "VfL OsnabrÃ¼ck",
    "spiele": 26,
    "punkte": 18,
    "tore": 24,
    "gegentore": 53,
    "tordifferenz": -29,
    "siege": 3,
    "unentschieden": 9,
    "niederlagen": 14,
    "letzte5": "NNSSN",
    "tendenz": [
      "N",
      "N",
      "S",
      "S",
      "N"
    ]
  }
]
